app/assets/javascripts/training/components/training_slide_handler.jsx

```diff
@@ -76,11 +87,27 @@ const TrainingSlideHandler = () => {
     }));
   };
 
+
+  const [isShown, setIsShown] = useState(false);
+
+  // this function checks whether alert should be shown or not.
+  const Alerthandler = () => {
+    if (routeParams.library_id === 'students') {
+    count += 1;
+    const clickingTime = new Date().getTime() - enteringTime;
+    if (count > 3 && clickingTime < 10000 && nooftimes < 2) {
```

Comment:
There are a lot of unnamed numbers here. The clickingTime comparison value, in particular, should be a well-named constant, and the other two values should probably also be constants with names that explain what they are for (eg, maxTooFastAlerts or similar).

Reply:
Thank you for your input, I'll fix this asap.